test/e2e/framework/framework.go

```diff
@@ -104,6 +111,27 @@ func (f *Framework) AfterEach() {
 	Expect(err).NotTo(HaveOccurred())
 }
 
+func (f *Framework) CertificateDurationValid(c *v1alpha1.Certificate, duration time.Duration) {
+	By("Verifying TLS certificate exists")
+	secret, err := f.KubeClientSet.CoreV1().Secrets(f.Namespace.Name).Get(c.Spec.SecretName, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	certBytes, ok := secret.Data[api.TLSCertKey]
+	if !ok {
+		Failf("No certificate data found for Certificate %q", c.Name)
+	}
+	cert, err := pki.DecodeX509CertificateBytes(certBytes)
+	Expect(err).NotTo(HaveOccurred())
+	By("Verifying that the duration is valid")
+	if cert.NotAfter.Sub(cert.NotBefore) != duration {
+		Failf("Expected duration of %s, got %s [NotBefore: %s, NotAfter: %s]", duration, cert.NotAfter.Sub(cert.NotBefore), cert.NotBefore.Format(time.RFC3339), cert.NotAfter.Format(time.RFC3339))
+	}
+}
+func (f *Framework) WaitForCertificateEvent(cert *v1alpha1.Certificate, reason string) {
+	By("Verifying that the certificate has an event with reason " + reason)
+	err := util.WaitForCertificateEvent(f.KubeClientSet, cert, reason, defaultTimeout)
+	Expect(err).NotTo(HaveOccurred())
+}
```

Comment:
Unused function?

Reply:
Right since we arent reporting events anymore for changes to the certificate duration this should have been removed